### PR TITLE
CNRFC rating curve #61

### LINKED ISSUE #61: CNRFC rating curve
### Task 
CNRFC web resources include JavaScript files specifying the construction of flow to stage rating curves for river forecast locations.  This task specifies adding support for extraction of this paired data in the `collect.cnrfc` module.

### Steps
1. Review available rating curve resources, specified by URLs of form https://www.cnrfc.noaa.gov/data/ratings/VNSC1_rating.js
2. Explore river forecast points displayed on CNRFC landing page: https://www.cnrfc.noaa.gov/ (see file download options to export river forecast locations vs. ensemble forecast points)
3. See `collect.cnrfc.gages` for additional resources on gages
4. Define `collect.cnrfc.cnrfc.get_rating_curve(cnrfc_id, other_arguments?)` function with additional arguments (if necessary) that returns paired flow and stage data parsed from the text of the rating curve URLs

### Testing Goals
- Ensure feature returns paired data for all valid river forecast locations
- Test that rating curve can be used to recreate a forecast stage hydrograph using the forecast flow series 

### Breaking Changes
None anticipated.

### Priority
Low

### PR DESCRIPTION
**Pull request by jinducil**
Closes #61 
Added features:
- Returns list of paired stage and flow data from URLs of form: https://www.cnrfc.noaa.gov/data/ratings/{cnrfc_id}_rating.js

### CODE REVIEW

**Review by narlesky — CHANGES_REQUESTED** [review]
Hi @jinducil - thanks for adding this!  I had an occasion to pull down these rating curves a couple weeks ago so thought I'd circle back around.  Here are my notes:

- I'd like to see this function refactored to use `requests`
- I also think you can refactor to drop the dependency on `BeautifulSoup`, since the structure of the file does not include XML or HTML
- consider combining the `map` expressions in lines 828-829 and 830-831
- please use single-quotes where possible
- Docstring please!

Thanks!

File: collect/cnrfc/cnrfc.py [hunk 1]
```diff
@@ -850,6 +850,47 @@ def get_forecast_csvdata(url):
...
+    Returns:
+        result (dict): query result dictionary with 'data' and 'info' keys
+    """
+    # retrieve data from URL
+    url = f'https://www.cnrfc.noaa.gov/data/ratings/{cnrfc_id}_rating.js'
+    response = requests.get(url)
+
+    # check if data exists
+    if response.status_code == 200:
+        raw_data = response.text.splitlines()[2:]
```
Comment (narlesky) [inline collect/cnrfc/cnrfc.py:868]: Since you are using `startswith` below, I don't think you need to explicitly specify the `[2:]`.

Comment (jinducil) [reply collect/cnrfc/cnrfc.py]: Resolved in [1a296a7](https://github.com/MBKEngineers/collect/pull/68/commits/1a296a7d646d7a758fced757353de4751efb9964)

**Review by narlesky — APPROVED** [review]
Thanks for adding this @jinducil !

**Pull request merged**